pkg/util/k8scmcache.go

```diff
@@ -0,0 +1,208 @@
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+	"sync"
+
+        "github.com/golang/glog"
+        "github.com/pkg/errors"
+
+        k8s "k8s.io/client-go/kubernetes"
+        "k8s.io/client-go/rest"
+        "k8s.io/client-go/tools/clientcmd"
+        "k8s.io/api/core/v1"
+        metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type K8sCMCache struct {
+	Client *k8s.Clientset
+	Namespace string
+
+}
+
+const (
+        defaultNamespace = "default"
+
+        cmLabel = "csi-metadata"
+        cmDataKey = "content"
+
+        csiMetadataLabelAttr = "com.ceph.ceph-csi/persister"
+)
+
+var (
+	k8sCMCacheMtx sync.Mutex
```

Comment:
Since we're not sharing data between cache entries anymore, we don't need to lock.